### DISCUSSION TITLE
what happens in case of Error: unable to heal object

### DISCUSSION THREAD

**Discussion by julienlau** [minio/minio#20557] [Q&A]
I got one node in a cluster showing a lot of read error like 
`Error: unable to heal object myapp/xxx.parquet`
This is not specific to a single object and I had a faulty drive replaced on this server, 

What happens to this object myapp/xxx.parquet once the heal error is logged ? 
Is there a retry on healing ?
Is it deleted from the bucket ?
It it put aside somewhere ?

More generally, what happens in case the `Objects Failed: ` counter is non null when running `mc admin heal myminio-local` ?

Regards
JL

**Comment by AlexZIX** [comment]
It means that you lost your object because it can't be healed and depends of how many drives in your MinIO setup, how many faulty drives you've replaced and when, which EC:N you use.

For ex. I have 16 drives and followng EC settings:

MINIO_STORAGE_CLASS_STANDARD=EC:6
MINIO_STORAGE_CLASS_RRS=EC:2

It means that for standard storage class I can lost up to 6 disks and for rrs - 2 disks. So if I lost more then few objects can't be healed because MinIO can't heal it from nothing. Also you should understand that healing is a long process. If you get 2 faulty drives, replaces it and got 3rd faulty drive in few hours while the data hasn't yet been restored, you can lost your objects.

**Comment by julienlau** [comment]
Thanks !
This is normal : all storage solutions can loose data and have a defined fault tolerance depending on the configuration.

What is bothering me with minio is that it loose data silently without any warning message.
There is an INFO message when minio attempt the healing but if the healing fails because there are too few shards left alive you never know.
This is very disappointing.
